flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/BufferedUpsertSinkFunctionTest.java

```diff
@@ -129,17 +140,23 @@
                 TimestampData.fromInstant(Instant.parse("2021-03-30T21:00:00Z")))
     };
 
+    private final boolean enableObjectReuse;
+
+    public BufferedUpsertSinkFunctionTest(boolean enableObjectReuse) {
+        this.enableObjectReuse = enableObjectReuse;
+    }
+
     @Test
     public void testWriteData() throws Exception {
-        MockedSinkFunction sinkFunction = new MockedSinkFunction();
+        MockedSinkFunction sinkFunction = new MockedSinkFunction(enableObjectReuse);
```

Comment:
You can make `MockedSinkFunction` non-static to avoid passing `enableObjectReuse` here and there.